Lib/test/string_tests.py

```diff
@@ -295,6 +283,25 @@ def test_index(self):
         else:
             self.checkraises(TypeError, 'hello', 'index', 42)
 
+        # For a variety of combinations,
+        #    verify that str.index() matches __contains__
+        #    and that the found substring is really at that location
+        charset = ['', 'a', 'b', 'c']
+        digits = 5
+        teststrings = self._get_teststrings(charset, digits)
+        for i in teststrings:
+            for j in teststrings:
+                if j in i:
+                    loc = i.index(j)
+                else:
+                    self.assertRaises(ValueError, i.index, j)
+                    loc = -1
+                r1 = (loc != -1)
+                r2 = j in i
+                self.assertEqual(r1, r2)
+                if loc != -1:
+                    self.assertEqual(i[loc:loc+len(j)], j)
```

Comment:
This can be simplified:
```suggestion
                if j in i:
                    loc = i.index(j)
                    self.assertGreaterEqual(loc, 0)
                    self.assertEqual(i[loc:loc+len(j)], j)
                else:
                    self.assertRaises(ValueError, i.index, j)
```